opennebula/resource_opennebula_virtual_network.go

```diff
@@ -383,29 +405,39 @@ func resourceOpennebulaVirtualNetworkCreate(d *schema.ResourceData, meta interfa
 	controller := meta.(*goca.Controller)
 	var vnc *goca.VirtualNetworkController
 
+	reservation_vnet := d.Get("reservation_vnet").(int)
+
 	// VNET reservation
-	if rvnet, ok := d.GetOk("reservation_vnet"); ok {
-		reservation_vnet := rvnet.(int)
-		reservation_name := d.Get("name").(string)
-		reservation_size := d.Get("reservation_size").(int)
-
-		if reservation_vnet < 0 {
-			return fmt.Errorf("Reservation VNET ID must be greater than 0!")
-		} else if reservation_size <= 0 {
-			return fmt.Errorf("Reservation size must be strictly greater than 0!")
+	if reservation_vnet > 0 {
```

Comment:
this would exclude the virtual network ID 0, at least, use -1 as a default value for reservation_vnet and check reservation_vnet > 0